packages/shared_preferences/shared_preferences_foundation/pigeons/messages.dart

```diff
@@ -20,6 +20,7 @@ abstract class UserDefaultsApi {
   void setValue(String key, Object value);
   // TODO(stuartmorgan): Make these non-nullable once
   // https://github.com/flutter/flutter/issues/97848 is fixed.
-  Map<String?, Object?> getAllWithPrefix(String prefix);
-  void clearWithPrefix(String prefix);
+  Map<String?, Object?> getAllWithParameters(
```

Comment:
You can just call these `getAll` and `clear` since we aren't contending with layers of previous versions we have to preserve.